### PR TITLE
Reducing nr of loops and if statements, added test file with few test…

### PR DESCRIPTION
**Pull request by rawadabu**
… function to get into testing

### CODE REVIEW

File: test.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import unittest
+from question_model import Question
+from data import question_data
+from quiz_brain import QuizBrain
+
+NR_QUESTIONS_EXPECTED = 12
```
Comment (yftacherzog) [inline test.py]:
```suggestion
NR_QUESTIONS_EXPECTED = len(question_bank)
```

File: test.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    def test_initialization(self):
+        self.assertEqual(self.quiz.question_number, NR_QUESTIONS_EXPECTED)
+        self.assertEqual(self.quiz.score, SCORE_EXPECTED)
+        self.assertEqual(len(self.quiz.questions_list), NR_QUESTIONS_EXPECTED)
+
+    def test_still_has_question(self):
+        self.assertTrue(self.quiz.still_has_question())
+        # Assuming the quiz has been set up with 12 questions, let's answer all of them
+        for _ in range(12):
```
Comment (yftacherzog) [inline test.py]:
```suggestion
        for _ in range(NR_QUESTIONS_EXPECTED):
```

File: test.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            self.quiz.next_question()
+        self.assertFalse(self.quiz.still_has_question())  # After answering all questions, there should be none left
+
+    def test_check_answer_correct(self):
```
Comment (yftacherzog) [inline test.py:29]: What happens if you have an empty quiz and you try to check an answer? I believe you're going to divide by zero.

File: test.py (hunk 1 above)
Comment (rawadabu) [reply test.py]: Done

File: test.py (hunk 2 above)
Comment (rawadabu) [reply test.py]: Done

File: test.py (hunk 3 above)
Comment (rawadabu) [reply test.py:29]: [394ebdf](https://github.com/rawadabu/quiz-game/pull/1/commits/394ebdf3217e45adbea2bcf9a0f60a0ba620a4b9) You are correct that there is a potential issue if the quiz is empty. However, the code has been updated and I think I've handled this scenario @yftacherzog

Comment (yftacherzog) [reply test.py:29]: I would have opted for eliminating the reason for the corner case to exist instead of adding a validation (another `if`, another potential bug, another required test).

In this case, `check_answer()` is suppose to check something that is related to a question on the quiz, but it's not actually doing that. It is instead checking arbitrary equality between strings, and this is why you can run it when there are no questions in the quiz.

If you instead have it as a method of `Question`, you won't have to verify that the question actually exists, as it would be impossible to call it without a question.

Comment (rawadabu) [reply test.py:29]: Got it, updating
